### DISCUSSION TITLE
Scroll a specific div to bottom

### DISCUSSION THREAD

**Discussion by notkojak** [seleniumbase/SeleniumBase#1571] [Q&A] [answered]
Hello, I want to scroll a div that has a list of items to the bottom, but this div is not the main content of the page but acts more like a menu.

I would like to know if it is possible to scroll a specific div to the bottom 

```
self.click('button.mBiMV')
    self.switch_to_frame_of_element('div.QAr02')
    print('element found')
    self.scroll_to_bottom()
    print('scroll')
    self.sleep(1000)
```
I tried that, but it's not scrolling to bottom
Thank you

**Comment by mdmintz** [comment, marked as answer]
I can't tell what you're trying to do after you scroll, but if the goal is to click on something afterward that is currently out-of-view, then you may be able to click it directly using ``self.js_click(SELECTOR)``:

```python
self.js_click(SELECTOR)
```

**Reply by notkojak** [reply]
Okay, this item is going to be useful to me, actually the list loads little by little, so I need to load it at least once completely by scrolling down 2 or 3 times, which I will do. 

I will then use your formula to click on the item.

But is there a javascript way to count like I do here the number of elements present
`if (len(self.find_elements('span.ogn1z')))>0:`

Because if I want to perform the click with javascript, I need to count the number of elements even outside the view

Thanks

**Reply by mdmintz** [reply]
There's ``self.find_elements(SELECTOR)`` and ``self.find_visible_elements(SELECTOR)``.

**Comment by hiyamandal** [comment]
I have the same issue right now. 
I am scrolling to the bottom of the page where I have a p element I want to assert. But the page is not scrolling and the test is failed with TestNotVisibleException.
The Code:
<img width="685" alt="image" src="https://user-images.githubusercontent.com/15275310/219649739-3ca57a3b-b45b-4a98-9803-9a595d08043c.png">
The error:
 seleniumbase.common.exceptions.TextNotVisibleException: 
E        Expected text substring {Integration} for {p.title} was not visible after 7 seconds!
E        (Actual string found was {Workshops})

**Reply by mdmintz** [reply]
Scrolling probably isn't going to change the text on an existing selector. When you search for a selector, SeleniumBase finds the first matching element. In your case, you may need to use a more-specific selector for your second element.
